Source/Csla.Channels.RabbitMq/RabbitMqChannelExtensions.cs

```diff
@@ -39,5 +41,25 @@ public static DataPortalClientOptions UseRabbitMqProxy(this DataPortalClientOpti
         });
       return config;
     }
+
+    /// <summary>
+    /// Configure the data portal server to use the RabbitMq channel.
+    /// </summary>
+    /// <param name="config"></param>
+    /// <param name="options"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static DataPortalServerOptions UseRabbitMqPortal(this DataPortalServerOptions config, Action<Channels.RabbitMq.RabbitMqPortalOptions>? options)
+    {
+      if (config is null)
+        throw new ArgumentNullException(nameof(config));
+
+      var portalOptions = new Channels.RabbitMq.RabbitMqPortalOptions();
+      options?.Invoke(portalOptions);
+
+      config.Services.AddScoped(_ => portalOptions);
+      config.Services.AddTransient<RabbitMqPortal>();
```

Comment:
We are sure this is only used in an asp.net core pipeline? Transient classes which implement `IDisposable` are a source of memory leak because the DI container tracks them until the holding "scope" gets disposed. In case of the root scope at the end of the application lifetime. Some information on that here https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#idisposable-guidance-for-transient-and-shared-instances

Reply:
This is actually intended for use in a console app, container, daemon, or Windows Service, not aspnetcore.

Reply:
So the transient service won't be disposed until
1. The user uses explicit scopes to resolve the dependency
2. The application shuts down
Can't we promote the lifetime to scoped/singleton and mention that a user _should_ use explicit scopes to avoid the memory leak?
Otherwise we could add a factory (as suggested by my link) which returns it and the user is now responsible for correct disposal.

Reply:
I switched to a factory model